src/components/button/styled.tsx

```diff
@@ -126,18 +126,10 @@ export const StyledButtonWrapper = styled.div<{ label: string }>`
   }};
 `
 
-export const StyledIcon = styled(({ name }) => <Icon className="button-icon" name={name} />)`
-  ${props => {
-    return css`
-      fill: ${buttonProps("buttonTextColor", props)};
-    `
-  }};
-`
-
-export const StyledButton = styled(({ label, icon, ...otherProps }) => (
+export const StyledButton = styled(({ label, icon, danger, ...otherProps }) => (
   <StyledButtonWrapper label={label}>
-    <button type="button" {...otherProps}>
-      {icon ? <StyledIcon name={icon} {...otherProps} /> : null}
+    <button type="button" danger={danger.toString()} {...otherProps}>
```

Comment:
Why is .toString() needed?

Reply:
As far as I've seen this was the error hitting since danger is boolean.

Reply:
I don't understand it still. Why we need the "danger" flag converted to string, what is happening overall? I get that there was an error, but what's the reason we need this logic?